Cargo.lock

```diff
@@ -2424,13 +2435,28 @@ version = "0.42.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "8e5180c00cd44c9b1c88adb3693291f1cd93605ded80c250a75d472756b4d071"
 dependencies = [
- "windows_aarch64_gnullvm",
- "windows_aarch64_msvc",
- "windows_i686_gnu",
- "windows_i686_msvc",
- "windows_x86_64_gnu",
- "windows_x86_64_gnullvm",
- "windows_x86_64_msvc",
+ "windows_aarch64_gnullvm 0.42.2",
+ "windows_aarch64_msvc 0.42.2",
+ "windows_i686_gnu 0.42.2",
+ "windows_i686_msvc 0.42.2",
+ "windows_x86_64_gnu 0.42.2",
+ "windows_x86_64_gnullvm 0.42.2",
+ "windows_x86_64_msvc 0.42.2",
```

Comment:
This looks like it's introducing a duplicate crate dependency.

Reply:
The only crate still depending on this older version is `dirs-sys 0.4.0`. The transitive dependency was updated in `dirs-sys 0.4.1` which would remove the duplicate dependency; however, that version also introduces a dependency on `option-ext` which is under the Mozilla Public License, a (weak) copyleft license that we currently do not have any dependency on (per the license checking logic I added in this PR). We would need to consider whether that is permissible for `zcashd` before upgrading.

Reply:
Also note that `terminfo` depends transitively on `dirs-sys 0.3`, which in turn depends on the `winapi` crate (which was the community-maintained API bindings to the Windows SDK before Microsoft started releasing their own API bindings under the `windows-sys` crate). So we already have effectively duplicate dependencies here, and they aren't going away any time soon (though hopefully over time).

Reply:
Filed https://github.com/dirs-dev/dirs-sys-rs/issues/23, but I'm not hopeful since the similar https://github.com/dirs-dev/dirs-sys-rs/issues/21 was closed as wontfix. The maintainer apparently does not care that they made an incompatible license change in a supposedly non-breaking release.

Reply:
Also filed https://github.com/dirs-dev/dirs-sys-rs/pull/24 to fix it.

Reply:
Issue and PR were silently closed. That's the developer's choice and they are welcome to it, but I'll start the process of migrating us away from this dependency.

Reply:
Opened https://github.com/zcash/librustzcash/pull/864.

Reply:
And we now depend on `zcash_proofs 0.12.1` which removes these duplicates.